pkg/api/server/v1alpha2/plugin/server.go

```diff
@@ -0,0 +1,121 @@
+package plugin
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"net"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/tektoncd/results/pkg/api/server/config"
+	"go.uber.org/zap"
+
+	"github.com/tektoncd/results/pkg/api/server/v1alpha2/auth"
+	"golang.org/x/oauth2"
+	"gorm.io/gorm"
+
+	pb3 "github.com/tektoncd/results/proto/v1alpha3/results_go_proto"
+
+	"k8s.io/client-go/transport"
+)
+
+// LogServer is the server for the log plugin server
+type LogServer struct {
+	pb3.UnimplementedLogsServer
+
+	IsLogPluginEnabled bool
+	staticLabels       string
+
+	config *config.Config
+	logger *zap.SugaredLogger
+	auth   auth.Checker
+	db     *gorm.DB
+	client *http.Client
+
+	forwarderDelayDuration time.Duration
+
+	queryLimit uint
+
+	queryParams map[string]string
+
+	// TODO: In future add support for non Oauth support
+	tokenSource oauth2.TokenSource
+
+	getLog getLog
+}
+
+// NewLogServer returns a plugin log server
+func NewLogServer(config *config.Config, logger *zap.SugaredLogger, auth auth.Checker, db *gorm.DB) (*LogServer, error) {
+	s := &LogServer{
+		config: config,
+		logger: logger,
+		auth:   auth,
+		db:     db,
+	}
+
+	s.logger.Debugf("LOGS_TYPE: %s", strings.ToLower(s.config.LOGS_TYPE))
+	// If the logs type is not Loki, we don't need to set up the LogPluginServer
```

Comment:
setLogPlugin() with also return true if the log type BlobLogType, so either the comment is outdated or the logic below.

Reply:
Yes. Comment is outdated.

Reply:
Updated this.